app/models/piece.rb

```diff
@@ -1,11 +1,44 @@
 class Piece < ActiveRecord::Base
-  belongs_to :game
-  belongs_to :user
-
-  def x_y_coords
-    x_y_coordinates = []
-    x_y_coordinates << x_position
-    x_y_coordinates << y_position
-    x_y_coordinates
-  end
+
+	belongs_to :user
+	belongs_to :game
+
+	# Lines 7-17 all part of STI: to break it disable line 6 or give it fake field name ~AMP
+	self.inheritance_column = :piece_type
+	scope :pawns, -> { where(piece_type: "pawn")}
+	scope :queens, -> { where(piece_type: "queen")}
+	scope :kings, -> { where(piece_type: "king")}
+	scope :rooks, -> { where(piece_type: "rook")}
+	scope :knights, -> { where(piece_type: "knight")}
+	scope :bishops, -> { where(piece_type: "bishop")}
+
+	def self.piece_types
+		%w(Pawn Queen King Rook Knight Bishop)
+	end
+
+	def x_y_coords
+		x_y_coordinates = []
+		x_y_coordinates << x_position
+		x_y_coordinates << y_position
+		x_y_coordinates
+	end
+
+	def is_captured? 
```

Comment:
Can just use Piece.first.captured? which automatically checks the value of a boolean member variable/field. Also if you use "where", I believe you have to pass it a Model, currently if you try to use this code as-is, it's broken.